### ISSUE TITLE
index out of range [0] with length 0 in `(*VecGroupChecker).GetNextGroup`

### ISSUE THREAD

**Issue by GaranR** [pingcap/tidb#52764]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
First execute the following valid.sql
[valid.txt](https://github.com/pingcap/tidb/files/15038430/valid.txt)

Then a crash occurs when executing the error.sql below
[error.txt](https://github.com/pingcap/tidb/files/15038435/error.txt)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
Expect no crashes

### 3. What did you see instead (Required)
```
runtime error: index out of range [0] with length 0
```
tidb.log:
```
[2024/04/17 11:57:25.569 +00:00] [ERROR] [executor.go:1845] ["resultPuller panicked"] [conn=408946054] [session_alias=] [recover="runtime error: index out of range [0] with length 0"] [stack="github.com/pingcap/tidb/pkg/executor.(*UnionExec).resultPuller.func1
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/executor.go:1845
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:914
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:114
github.com/pingcap/tidb/pkg/executor/internal/vecgroupchecker.(*VecGroupChecker).GetNextGroup
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/internal/vecgroupchecker/vec_group_checker.go:477
github.com/pingcap/tidb/pkg/executor/aggregate.(*StreamAggExec).consumeOneGroup
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/aggregate/agg_stream_executor.go:131
github.com/pingcap/tidb/pkg/executor/aggregate.(*StreamAggExec).Next
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/aggregate/agg_stream_executor.go:109
github.com/pingcap/tidb/pkg/executor/internal/exec.Next
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/internal/exec/executor.go:283
github.com/pingcap/tidb/pkg/executor.(*LimitExec).Next
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/executor.go:1369
github.com/pingcap/tidb/pkg/executor/internal/exec.Next
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/internal/exec/executor.go:283
github.com/pingcap/tidb/pkg/executor.(*UnionExec).resultPuller
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/executor.go:1875"]
[2024/04/17 11:57:25.570 +00:00] [INFO] [conn.go:1124] ["command dispatched failed"] [conn=408946054] [session_alias=] [connInfo="id:408946054, addr:10.0.2.1:49688 status:10, collation:utf8mb4_0900_ai_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="(check error.sql above)"] [txn_mode=PESSIMISTIC] [timestamp=449145745054105611] [err="runtime error: index out of range [0] with length 0
github.com/pingcap/tidb/pkg/executor.(*UnionExec).resultPuller.func1
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/executor.go:1846
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:914
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:114
github.com/pingcap/tidb/pkg/executor/internal/vecgroupchecker.(*VecGroupChecker).GetNextGroup
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/internal/vecgroupchecker/vec_group_checker.go:477
github.com/pingcap/tidb/pkg/executor/aggregate.(*StreamAggExec).consumeOneGroup
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/aggregate/agg_stream_executor.go:131
github.com/pingcap/tidb/pkg/executor/aggregate.(*StreamAggExec).Next
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/aggregate/agg_stream_executor.go:109
github.com/pingcap/tidb/pkg/executor/internal/exec.Next
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/internal/exec/executor.go:283
github.com/pingcap/tidb/pkg/executor.(*LimitExec).Next
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/executor.go:1369
github.com/pingcap/tidb/pkg/executor/internal/exec.Next
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/internal/exec/executor.go:283
github.com/pingcap/tidb/pkg/executor.(*UnionExec).resultPuller
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/executor.go:1875
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1650"]
```

### 4. What is your TiDB version? (Required)
```
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                 |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v7.5.1
Edition: Community
Git Commit Hash: 7d16cc79e81bbf573124df3fd9351c26963f3e70
Git Branch: heads/refs/tags/v7.5.1
UTC Build Time: 2024-02-27 14:28:32
GoVersion: go1.21.6
Race Enabled: false
Check Table Before Drop: false
Store: tikv |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```
<!-- Paste the output of SELECT tidb_version() -->

We are the BASS team from the School of Cyber Science and Technology at Beihang University. Our main focus is on system software security, operating systems, and program analysis research, as well as the development of automated program testing frameworks for detecting software defects. Using our self-developed database vulnerability testing tool, we have identified the above-mentioned vulnerabilities in TiDB that may lead to database crashes.

**Comment by zimulala** [comment]
Thanks for your report!
We fixed this issue in v8.2.0, and the corresponding PR is https://github.com/pingcap/tidb/pull/53874.
You can try it. I will close this issue for now. If there are any related questions, you can continue to give feedback.

**Comment by XuHuaiyu** [comment]
dup with https://github.com/pingcap/tidb/issues/53867
